File: src/ELibrary.Frontend/src/app/modules/shared/domain/dtos/library/author/authorResponse.ts
import { Author } from "../../../..";

export interface AuthorResponse {
    id: number;
    name: string;
    lastName: string;
    dateOfBirth: Date;
}
export function mapAuthorResponseToAuthor(response: AuthorResponse): Author {
    return {
        id: response.id,
        name: response.name,
        lastName: response.lastName,
        dateOfBirth: new Date(response.dateOfBirth)
    }
}

File: src/ELibrary.Backend/LibraryShopEntities/Domain/Dtos/Library/AuthorResponse.cs
namespace LibraryShopEntities.Domain.Dtos.Library
{
    public class AuthorResponse
    {
        public int Id { get; set; }
        public string? Name { get; set; }
        public string? LastName { get; set; }
        public DateTime DateOfBirth { get; set; }
    }
}
